core/utils.c

```diff
@@ -252,64 +300,189 @@ size_t utils_floatToText(double data,
                          uint8_t * string,
                          size_t length)
 {
-    size_t intLength;
-    size_t decLength;
-    int64_t intPart;
+    uint64_t intPart;
     double decPart;
+    double noiseFloor;
+    size_t res;
+    size_t head = 0;
+    int exp = 0;
+    uint8_t expLen = 0;
+    int precisionFactor = 1; /* Adjusts for inaccuracies caused by power of 10 operations. */
 
-    if (data <= (double)INT64_MIN || data >= (double)INT64_MAX) return 0;
+    if (!length || !string) return 0;
 
-    intPart = (int64_t)data;
-    decPart = data - intPart;
-    if (decPart < 0)
+    if (data < 0)
     {
-        decPart = 1 - decPart;
+        string[head++] = '-';
+        data = -data;
     }
-    else
+
+    /* Handle special cases */
+    if (data < DBL_MIN)
     {
-        decPart = 1 + decPart;
+        /* Intentionally not distinguishing between +0.0 and -0.0. */
+        if (length < 3) return 0;
+        string[0] = '0';
+        string[1] = '.';
+        string[2] = '0';
+        if (length > 3) string[3] = '\0';
+        return 3;
     }
-
-    if (decPart <= 1 + FLT_EPSILON)
+    else if (data > DBL_MAX )
```

Comment:
Shouldn't this never be true ?

Reply:
DBL_MAX is the largest finite double value. Infinity will be larger. Infinity needs special handling here or later loops will never terminate.